invenio_rdm_records/services/services.py

```diff
@@ -128,3 +145,289 @@ def oai_result_item(self, identity, oai_record_source):
+            )
+        else:
+            values = [
+                v.value for v in RecordDeletionStatusEnum if v.value != deletion_status
+            ]
+            filter = dsl.Q(
+                "bool", **{"must_not": {"terms": {"deletion_status": values}}}
+            )
+
+        if extra_filter := kwargs_dict.get("extra_filter", None):
+            filter = filter & extra_filter
+
+        kwargs_dict["extra_filter"] = filter
+        return kwargs_dict
+
+    def search(
+        self, identity, params=None, search_preference=None, expand=False, **kwargs
+    ):
+        """Search for published records matching the querystring."""
+        status = RecordDeletionStatusEnum.PUBLISHED.value
+        kwargs = self._add_deletion_filter(kwargs, status)
+        return super().search(identity, params, search_preference, expand, **kwargs)
+
+    def search_deleted(
+        self, identity, params=None, search_preference=None, expand=False, **kwargs
+    ):
+        """Search for soft-deleted records matching the querystring."""
+        status = RecordDeletionStatusEnum.DELETED.value
+        kwargs = self._add_deletion_filter(kwargs, status)
+        return super().search(identity, params, search_preference, expand, **kwargs)
+
+    def search_marked_for_purge(
+        self, identity, params=None, search_preference=None, expand=False, **kwargs
+    ):
+        """Search for records marked for purge, matching the querystring."""
+        status = RecordDeletionStatusEnum.MARKED.value
+        kwargs = self._add_deletion_filter(kwargs, status)
+        return super().search(identity, params, search_preference, expand, **kwargs)
+
+    def search_drafts(
+        self, identity, params=None, search_preference=None, expand=False, **kwargs
+    ):
+        """Search for drafts that have not been marked as deleted."""
+        # because drafts don't have a 'deletion_status', a simple positive filter
+        # won't work in cases where records and drafts are mixed...
+        # instead, we exclude all the 'wrong' values here
+        status = RecordDeletionStatusEnum.PUBLISHED.value
+        kwargs = self._add_deletion_filter(kwargs, status, positive_filter=False)
+        return super().search_drafts(
+            identity,
+            params=params,
+            search_preference=search_preference,
+            expand=expand,
+            **kwargs,
+        )
+
+    def search_versions(
+        self, identity, id_, params=None, search_preference=None, expand=False, **kwargs
+    ):
+        """Search versions of a record that aren't deleted."""
+        return super().search_versions(
+            identity,
+            id_,
+            params=None,
+            search_preference=None,
+            expand=False,
+            **kwargs,
+        )
+
+    #
+    # Base methods, extended with handling of deleted records
+    #
+    def read(self, identity, id_, expand=False, read_deleted=False):
```

Comment:
```suggestion
    def read(self, identity, id_, expand=False, with_deleted=False):
```
I think in other places we use the `with_deleted` convention e.g https://github.com/inveniosoftware/invenio-records/blob/master/invenio_records/api.py#L370